components/ui/button/utils/helpers.ts

```diff
@@ -1,17 +1,24 @@
 import * as React from "react";
 
-import { ButtonColors, ButtonProps, ButtonSizes } from "./types";
+import {
+  ButtonColors,
+  ButtonIconPositions,
+  ButtonProps,
+  ButtonSizes,
+} from "./types";
 
 import clsx from "clsx";
 
 export const disabledStyles = "disabled:cursor-not-allowed disabled:opacity-75";
 
-export function buttonBlockStyles(block?: boolean) {
-  if (block) {
-    return "flex";
-  }
-
-  return "inline-flex";
+export function buttonBlockStyles(
+  block?: boolean,
+  iconPosition: ButtonIconPositions = "left",
+) {
+  return [
+    block ? "flex" : "inline-flex",
+    iconPosition === "right" ? "flex-row-reverse" : "flex-row",
```

Comment:
Don't know if doing this is better than reversing the rendering in the component itself, but this reduces the chance of code copy/paste. Though I guess converting `renderButtonIcon` to a general-purpose `renderButtonContent` might also work, but...